test/e2e/imagecredentialprovider/credential_provider.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentialprovider
+
+import (
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+// SIGDescribe annotates the test with the SIG label.
+func SIGDescribe(text string, body func()) bool {
+	return ginkgo.Describe("[sig-cloud-provider][sig-node][sig-auth] "+text, body)
+}
+
+/*
+These tests assumes external gcp credential provider is installed and configured on k8s node.
+however they won't fail unless in tree gcp credential provider is removed or disabled.
+Currently to make sure external credential provider is used for tests both below feature gates needs to be enabled in kubelet
+KubeletCredentialProviders, DisableKubeletCloudCredentialProviders
+*/
```

Comment:
the e2e framework was trying to avoid cloud-provider dependency, cc: @BenTheElder 

anyway, if this is allowed, it should have a `e2eskipper.SkipUnlessProviderIs("gce")` statement to skip based on the required provider

Reply:
`e2eskipper.SkipUnlessProviderIs("gce")` won't work here, these tests  meant to be run with external gcp cloud provider where --provider=skeleton is passed

Reply:
I think setting the test provider to gce but setting DisableKubeletCloudCredentialProviders=true should be sufficient for now.  Would that work?

Reply:
to run external gcp credential provider we need to run external gcp cloud provider, if we look at the cloud-provider-gcp repo the CI job which runs e2e tests does not set gce as provider, for out of tree providers i guess `skeleton` is used. 
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/cloud-provider-gcp/319/cloud-provider-gcp-e2e-full/1499045151052402688/build-log.txt

so to run these tests i would add a CI job [here](KubeletCredentialProviders) with `DisableKubeletCloudCredentialProviders=true & KubeletCredentialProviders=true` , need to add in cloud-provider-gcp jobs as scripts to install and provision cloud+credential is already available